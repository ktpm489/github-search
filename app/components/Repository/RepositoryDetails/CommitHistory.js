import React from 'react';
import PropTypes from 'prop-types';
import {
  FlatList,
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform,
} from 'react-native';
import { WebBrowser } from 'expo';

const handleOpenCommit = async (commitUrl) => {
  await WebBrowser.openBrowserAsync(commitUrl);
};

export const TouchableCommit = ({ children, commitUrl }) =>
  Platform.OS === 'ios' ? (
    <TouchableOpacity onPress={() => handleOpenCommit(commitUrl)}>
      {children}
    </TouchableOpacity>
  ) : (
    <TouchableNativeFeedback onPress={() => handleOpenCommit(commitUrl)}>
      {children}
    </TouchableNativeFeedback>
  );

TouchableCommit.propTypes = {
  commitUrl: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

const CommitHistory = ({ edges }) => (
  <FlatList
    style={{ alignSelf: 'stretch' }}
    data={edges}
    renderItem={({
      item: {
        node: { message, oid, commitUrl },
      },
    }) => (
      <TouchableCommit commitUrl={commitUrl}>
        <View key={oid}>
          <Text style={styles.commitId}>{oid.substring(0, 7)}</Text>
          <Text>{message}</Text>
        </View>
      </TouchableCommit>
    )}
    keyExtractor={(item) => item.node.oid}
  />
);

CommitHistory.propTypes = {
  edges: PropTypes.arrayOf(
    PropTypes.shape({
      node: {
        message: PropTypes.string,
        oid: PropTypes.string,
        commitUrl: PropTypes.string,
      },
    })
  ),
};

export default CommitHistory;

const styles = StyleSheet.create({
  commitId: {
    textDecorationLine: 'underline',
    fontSize: 15,
    fontWeight: '700',
    color: '#222',
  },
});
