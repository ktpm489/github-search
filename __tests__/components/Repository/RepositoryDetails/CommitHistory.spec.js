import React from 'react';
import renderer from 'react-test-renderer';

import CommitHistory, {
  TouchableCommit,
} from '../../../../app/components/Repository/RepositoryDetails/CommitHistory';

describe('<CommitHistory />', () => {
  const edges = [
    {
      node: {
        oid: '12345',
        message: 'Commit me please!',
        commitUrl: 'mycommit.github.io',
      },
    },
    {
      node: {
        oid: '54321',
        message: 'Maybe commit me?',
        commitUrl: 'notmycommit.github.io',
      },
    },
  ];

  it('renders correctly with all required props', () => {
    const tree = renderer.create(<CommitHistory edges={edges} />);
    expect(tree.toJSON()).toMatchSnapshot();

    const testInstance = tree.root;
    expect(testInstance.findAllByType(TouchableCommit)).toHaveLength(2);
  });
});
