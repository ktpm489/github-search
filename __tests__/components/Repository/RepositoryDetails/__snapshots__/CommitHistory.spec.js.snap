// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitHistory /> renders correctly with all required props 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "node": Object {
          "commitUrl": "mycommit.github.io",
          "message": "Commit me please!",
          "oid": "12345",
        },
      },
      Object {
        "node": Object {
          "commitUrl": "notmycommit.github.io",
          "message": "Maybe commit me?",
          "oid": "54321",
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#222",
                "fontSize": 15,
                "fontWeight": "700",
                "textDecorationLine": "underline",
              }
            }
          >
            12345
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Commit me please!
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#222",
                "fontSize": 15,
                "fontWeight": "700",
                "textDecorationLine": "underline",
              }
            }
          >
            54321
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Maybe commit me?
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
